File: minimum-swaps-to-group-all-1s-together-ii/solution.py
class Solution:
    def minSwaps(self, nums: List[int]) -> int:
        c=nums.count(1)
        n=len(nums)
        i=0
        cur=0
        ans=c
        for j in range(n*2):
            cur+=nums[j%n]
            while j-i+1>c:
                cur-=nums[i%n]
                i+=1
            if j-i+1==c:
                ans=min(ans,c-cur)
        return ans

                

File: minimum-swaps-to-group-all-1s-together-ii/solution.cpp
class Solution {
public:
    int minSwaps(vector<int>& nums) {
        int c = 0;
        for (int &x:nums)
            c+=x;
        int n = nums.size();
        int i = 0;
        int cur = 0;
        int ans = c;
        for (int j = 0; j < n * 2; ++j) {
            cur += nums[j % n];
            while (j - i + 1 > c) {
                cur -= nums[i % n];
                i++;
            }
            if (j - i + 1 == c) {
                ans = min(ans, c - cur);
            }
        }
        return ans;
    }
};